Implement ddah test

frankcbw opened the pull request:
Implemented test case "getting ddahs for one session will not return ddahs for another session"

siefkenj committed:
Fix position list in instructor view to be non-editable

siefkenj committed:
Add non-editable version of offer table

siefkenj committed:
Added Ddah routes to /api/v1/instructor prefix

siefkenj committed:
Stubbed tests for instructor Ddah routes.

frankcbw committed:
implemented ddah test

frankcbw committed:
implemented ddah test

siefkenj committed:
Merge pull request #485 from siefkenj/instructor-view

Instructor view fixes

Altair59 committed:
[frontend] implement tests for updating offer when emailed (#478)

* implemented tests for updating offer when emailed

siefkenj committed:
[backend] Add Reporting Tag Routes (#491)

* Add reporting tag routes and test stubs

siefkenj committed:
[backend] Add `instructor/instructors` POST route (#492)

Adds the ability for an instructor to update their name/email, but not their UTORid. Also, they cannot update the information of any other instructor.

frankcbw committed:
Finished ddah test

frankcbw committed:
Finished ddah test

frankcbw committed:
Added comments

frankcbw committed:
Added comments

Altair59 reviewed (commented):
As I understand, this code checks that every ddah fetched from one session is associated to one valid assignment from this session, thus can fulfill the test's goal to see if any ddah is from another session.

Accelsnow commented:
It feels weird that the original test suite creates a global ddah object for all test cases, which creates a layer of dependency among all test cases. Is there a way to make each of the ddah tests independent?

frankcbw committed:
Update styling and comment

frankcbw committed:
Update styling and comment

frankcbw commented:
One potential issue:

Under test "getting ddahs for one session will not return ddahs for another session", I tried to firstly add more ddahs with different assignments'ID, as below:

```
it("getting ddahs for one session will not return ddahs for another session", async () => {
        const newDdah = {
            assignment_id: assignments[1].id,
            duties: [
                {
                    order: 2,
                    hours: 25,
                    description: "Marking the midterm",
                },
                {
                    order: 1,
                    hours: 4,
                    description: "Initial training",
                },
                {
                    order: 3,
                    hours: 40,
                    description: "Running tutorials",
                },
            ],
        };

        const resp0 = await apiPOST(`/admin/ddahs`, newDdah);
        expect(resp0).toHaveStatus("success");

        // retrieve all ddahs for the session
        resp1 = await apiGET(`/admin/sessions/${session.id}/ddahs`);
        expect(resp1).toHaveStatus("success");
        const allDdah = resp1.payload;

        // retrieve all the assignments for the seesion
        const resp2 = await apiGET(`/admin/sessions/${session.id}/assignments`);
        expect(resp2).toHaveStatus("success");
        const allAssignmentId = resp2.payload.map((y) => y.id);

        // check that all the ddahs are associated with an assignment from the session
        allDdah.forEach((d) => {
            expect(allAssignmentId).toContain(d.assignment_id);
        });
    });
```
And it fails only MockAPI only with error message:
```
● Mock API tests › DDAH tests › create ddah from assignment route

    expected API response to have {"status": "success"} but received {"message": "Error: Duplicate entry exists for property assignment_id with value 3", "status": "error"}

      197 |             newDdah
      198 |         );
    > 199 |         expect(resp1).toHaveStatus("success");
          |                       ^
      200 |         expect(resp1.payload).toMatchObject({
      201 |             ...newDdah,
      202 |             assignment_id: newAssignment.id,
```

Without the insertion at the beginning, both API and Mock passed.

siefkenj commented on frontend/src/tests/ddah-tests.js:
```diff
@@ -88,9 +88,22 @@ export function ddahTests(api) {
+        // check that all the ddahs are associated with an assignment from the session
+        allDdah.forEach((d) => {
+            expect(allAssignmentId).toContain(d.assignment_id);
+        });
```
The default seed data only has one session and so only puts DDAHs in for that session. You need to create another session, a new assignment for that session, and a DDAH for that assignment. Then make sure fetching the DDAHs for the seed session and the newly created session return different objects.

reidka committed:
Create LICENSE

Used the MIT template from GitHub

siefkenj committed:
Merge pull request #494 from uoft-tapp/add-license-1

Create LICENSE

siefkenj committed:
Update Rails to 6.1.1 (#495)

* Update Rails to 6.1.1
* Update Ruby to 2.6.6

frankcbw committed:
Updated approach comparing ddahs from two sessions

frankcbw committed:
Updated approach comparing ddahs from two sessions

frankcbw committed:
remove global resp; add local resp

frankcbw committed:
remove global resp; add local resp

frankcbw committed:
changed test logic

frankcbw committed:
changed test logic

frankcbw committed:
Implement applicant test.  (#480)

* Update applicant tests

* remove applicant failing test from the mock api tests. The mock api doesn't need to handle all the edge cases that the real api does.

siefkenj commented on frontend/src/tests/ddah-tests.js:
```diff
@@ -77,20 +79,91 @@ export function ddahTests(api) {
+    it("getting ddahs for one session will not return ddahs for another session", async () => {
+        // first create a new session filled with new position, assignment, and contract.
+        const newSession = await addSession(
+            { apiGET: apiGET, apiPOST: apiPOST },
```
You can just pass `api`

siefkenj commented on frontend/src/tests/ddah-tests.js:
```diff
@@ -77,20 +79,91 @@ export function ddahTests(api) {
+            { apiGET: apiGET, apiPOST: apiPOST },
+            { contract_templates: true }
+        );
+        let resp1 = await api.apiGET(
```
No need to call it `resp1` if there is no `resp2`

siefkenj commented on frontend/src/tests/ddah-tests.js:
```diff
@@ -77,20 +79,91 @@ export function ddahTests(api) {
+            end_date: newSession.end_date,
+            contract_template_id: newContract.id,
+        };
+        resp1 = await api.apiPOST(
```
`apiPOST` is in the global namespace. No need to use `api.apiPOST`

osamaramihafez committed:
initial implementation for assignment insertion with null dates (#481)

Test that when creating an assignment with null start and end dates, the start/end dates from the position are inherited.

frankcbw committed:
fixed ddah test styling

frankcbw committed:
fixed ddah test styling

frankcbw committed:
Commit to rerun the unit test on travis

frankcbw committed:
Merge branch 'ddah-tests' into local ddah-tests

osamaramihafez commented:
Why are many commits from other branches appearing as changes on this branch? It makes it seem like there are too many file changes.

siefkenj commented:
> Why are many commits from other branches appearing as changes on this branch? It makes it seem like there are too many file changes.

You need to merge `upstream/master` into your branch

frankcbw commented:
That is what happened when I tried to fetch and merge from `upstream/master`. I am not sure if it is supposed to show such many fille changes, because It does make it difficult for reviewing.

frankcbw commented:
Also, have fixed issue #493
